native/src/extension/on_primitive.rs

```diff
@@ -53,9 +53,18 @@ pub fn write_variant_map(primitive: &mut Primitive, tag_to_ix: &HashMap<Tag, usi
             let mut tag_vec: Vec<Tag> = tags.iter().cloned().collect();
             // order tags deterministically
             tag_vec.sort_unstable();
+
+            let variants = tag_vec
+                .iter()
+                .map(|tag| {
+                    let (&variant_ix, _) = variant_ix_lookup.iter().find(|(_k, v)| v == &tag).unwrap();
+                    variant_ix as u32
+                })
+                .collect();
```

Comment:
Could you do a sort on the variants here?  Now that they are exposed as indices, probably just a bit cleaner if they are in index order

```rust
let mut variants = ...
variants.sort_unstable();
```

Reply:
Updated!